package/run.sh

```diff
@@ -20,7 +20,28 @@ else
     FORCE_RESTART=false
 fi
 
-env "INSTALL_K3S_FORCE_RESTART=${FORCE_RESTART}" "INSTALL_K3S_SKIP_DOWNLOAD=true" "INSTALL_K3S_SKIP_SELINUX_RPM=true" "INSTALL_K3S_SELINUX_WARN=true" installer.sh $@
+
+# work around for https://github.com/k3s-io/k3s/issues/2306
+# depends on journald being available on first boot, so this may not work on all providers (e.g. Digital Ocean)
+DELAY=10
+while ! env "INSTALL_K3S_FORCE_RESTART=${FORCE_RESTART}" "INSTALL_K3S_SKIP_DOWNLOAD=true" "INSTALL_K3S_SKIP_SELINUX_RPM=true" "INSTALL_K3S_SELINUX_WARN=true" installer.sh $@
+do
+    # Only use the logs for the latest restart of k3s, otherwise errors encountered after an ETCD join error will not be reported properly.
+    journalctl -u k3s.service > k3s-service.txt
```

Comment:
If I'm not mistaken, this is always going to error out on a worker node if the worker node installer fails, because the unit on a worker node is called `k3s-agent.service`. See: https://github.com/k3s-io/k3s/blob/454440f9a3b585bd038293f9de5acb470380c3fa/install.sh#L188-L199

You can check for this via the environment variable `INSTALL_K3S_EXEC` as per: https://github.com/rancher/rancher/blob/0811ad05c011e7becea052b107d1b62ce69e22d8/pkg/provisioningv2/rke2/planner/instructions.go#L58

Reply:
Good to know! I've updated the script to use the correct service based off of the presence of the `INSTALL_K3S_EXEC` environment variable